src/EditorFeatures/Core/Structure/AbstractStructureTaggerProvider.cs

```diff
@@ -67,37 +67,146 @@ protected override bool ComputeInitialTagsSynchronously(ITextBuffer subjectBuffe
+                }
+
+                // For the purposes of collapsing, extern aliases are grouped with usings, so
+                // we need to check for them too
+                if (line.StartsWith(index, "extern alias", ignoreCase: false))
+                    return true;
+
+                // For C# there are 5 types of statements that start with "using", but we only want to know about 2:
+                // Want to find:
+                //     1. using Y;
+                //     2. using X = Y;
+                //     3. using static X;
+                // Don't want to find:
+                //     4. using var X = Y;
+                //     5. using ({var X = Y|X = Y|X})
+                //
+                // So any line that starts with "using", ends with ";", doesn't have "(", and has
+                // 2 or 4 words.
+
+                // We expect the using keyword first
+                if (!line.StartsWith(index, "using", ignoreCase: false))
+                    return false;
+
+                // Skip past "using"
+                index += 5;
+
+                // Skip whitespace after using keyword
+                Skip(text, ref index, lineEnd, whitespace: true);
+
+                // If we find a static then this can only be case 3
+                if (line.StartsWith(index, "static", ignoreCase: false))
+                    return true;
+
+                // If we find an open paren then this is case 4
+                if (index >= lineEnd || text[index] == '(')
+                    return false;
+
+                // Skip the next word. It could be a namespace, alias, type name, or "var"
+                Skip(text, ref index, lineEnd, whitespace: false);
+
+                // Skip whitespace after that identifier
+                Skip(text, ref index, lineEnd, whitespace: true);
+
+                // Now we either need to find a semicolon, for case 1, or an equals sign, for case 3
+                if (index < lineEnd && (text[index] == ';' || text[index] == '='))
+                    return true;
+
+                // Otherwise it must be case 5
                 return false;
+            }
 
-                static bool StartsWithRegionTag(ITextSnapshotLine line)
+            static void Skip(ITextSnapshot text, ref int index, int lineEnd, bool whitespace)
+            {
+                for (; index < lineEnd; index++)
                 {
-                    var start = line.GetFirstNonWhitespacePosition();
-                    return start != null && line.StartsWith(start.Value, "#region", ignoreCase: true);
+                    var isWhitespace = text[index] is ' ' or '\t';
+                    if (isWhitespace != whitespace)
+                        return;
+                    // If we're skipping non-whitespace we want the caller to know about the
+                    // end of the statement
+                    if (text[index] == ';')
+                        return;
                 }
             }
         }
```

Comment:
I'm torn on this.

On one hand, I think it's interesting to think about whether your "near 100%" intuition holds in a post-global-usings world. I just did a search in my (granted, small) side project for the word "using" and I get:
<img width="183" alt="image" src="https://user-images.githubusercontent.com/754264/166081518-c114d5c6-0046-4771-9ea2-f0fbe7856c0e.png">

And no, I don't have _that_ many enums 😛
I even wrote a source generator for someone who wanted automatic global usings for every possible namespace from every reference, so that he never has to see a using statement.

On the other hand, I don't know how large the number of users would be that turn the option on, so I'm not against just always doing it, and wait for feedback. Certainly in my project even if the majority of files won't need it, they're also not huge. And my friend with the source generator probably won't have the setting turns on either 🤷‍♂️

Maybe simplifying to just "does the line start with 'using'" is a decent compromise, if you think this algorithm is too much.

Reply:
Yeah, this only helps the "is a very large file... but also has absolutely no usings in it" case.  That seems practically impossible to hit in practice.  WHat users have huge files, but also have absolutely no usings at all to bring in external symbols?

Reply:
I'm sure there are horrors out there, we just have to hope we don't have to look at them :)